static/js/ProfileDetail.tsx

```diff
@@ -0,0 +1,221 @@
+import React, { FC } from "react";
+import { useParams } from "react-router-dom";
+import { useQuery } from "@tanstack/react-query";
+import { fetchProfile } from "./api/profiles";
+import DeleteProfileBtn from "./buttons/profiles/DeleteProfileBtn";
+import EditProfileBtn from "./buttons/profiles/EditProfileBtn";
+import OpenProfileListBtn from "./buttons/profiles/OpenProfileListBtn";
+import BaseLayout from "./components/BaseLayout";
+import NotificationRow from "./components/NotificationRow";
+import { queryKeys } from "./util/queryKeys";
+import useNotification from "./util/useNotification";
+import {
+  CodeSnippet,
+  CodeSnippetBlockAppearance,
+  Col,
+  Row,
+} from "@canonical/react-components";
+import { LxdDiskDevice, LxdNicDevice } from "./types/device";
+
+const ProfileDetail: FC = () => {
+  const { name } = useParams<{ name: string }>();
+
+  if (!name) {
+    return <>Missing name</>;
+  }
+
+  const notify = useNotification();
+
+  const { data: profile, error } = useQuery({
+    queryKey: [queryKeys.profiles, name],
+    queryFn: () => fetchProfile(name),
+  });
+
+  if (error) {
+    notify.failure("Could not load profile details.", error);
+  }
+
+  if (!profile) {
```

Comment:
I think we can also end in this branch, when the query is loading. Generally we miss the loading state in our components.

Reply:
Right. I remembered that `useQuery` gives us the `isLoading` flag, so I used it to differentiate the loading case from the case where the profile details could not be loaded.
It is a temporary solution, and we should discuss how to visually present the loading state to the user in a nice way, but I think that for the scope of this PR it should be enough - and will also serve as a reminder for a future PR to generalise and improve this behaviour.